CustomLookup/CustomEntitySearch.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
```

Comment:
@ do these files need MSFT copyright headers?

Reply:
I think for open source projects, it's not required (see release checklist on the internal open source site). Doesn't hurt to add it though, I'll check for more authoritative advice and do it across the repo if it's required, no need to address it as part of this PR.

Reply:
Doesn't look like we *need* copyright headers, but it's generally considered good practice. See for instance https://forums.dotnetfoundation.org/t/file-headers-and-copyright-statements/1276 for the .NET Foundation's stance on the issue. I'll add notices throughout the repo.